c_mcai_worker_sdk/src/worker.rs

```diff
@@ -237,80 +294,272 @@ pub fn get_worker_parameters() -> Vec<Parameter> {
-            loop {
-              let cur_ptr = *ptr.offset(offset);
-              if cur_ptr.is_null() {
-                break;
-              }
-
-              output_paths.push(get_c_string!(cur_ptr));
-
-              libc::free(cur_ptr as *mut libc::c_void);
-              offset += 1;
-            }
-
-            if offset > 0 {
-              libc::free(ptr as *mut libc::c_void);
-            }
-          }
+  let worker_lib = libloading::Library::new(library).map_err(|error| {
+    MessageError::RuntimeError(format!(
+      "Could not load worker dynamic library: {:?}",
+      error
+    ))
+  })?;
 
-          // Retrieve message as string and free pointer
-          let mut message = "".to_string();
-          if !message_ptr.is_null() {
-            message = get_c_string!(message_ptr);
-            libc::free(message_ptr as *mut libc::c_void);
-          }
+  unsafe {
+    if let Ok(init_func) =
+      get_library_function::<libloading::Symbol<InitFunc>>(&worker_lib, constants::INIT_FUNCTION)
+    {
+      init_func(logger);
+    }
+  }
 
-          ProcessReturn::new(return_code, &message).with_output_paths(output_paths)
-        }
-        Err(error) => ProcessReturn::new_error(&format!(
+  Ok(())
+}
+
+#[cfg(feature = "media")]
+pub fn call_worker_init_process(
+  parameters: CWorkerParameters,
+  format_context: Arc<Mutex<FormatContext>>,
+) -> Result<Vec<usize>> {
+  let library = get_library_file_path();
+  debug!("Call worker process from library: {}", library);
+
+  let worker_lib = libloading::Library::new(library).map_err(|error| {
+    MessageError::RuntimeError(format!(
+      "Could not load worker dynamic library: {:?}",
+      error
+    ))
+  })?;
+
+  unsafe {
+    let init_process_func: libloading::Symbol<InitProcessFunc> =
+      get_library_function(&worker_lib, constants::INIT_PROCESS_FUNCTION).map_err(|error| {
+        MessageError::RuntimeError(format!(
           "Could not access {:?} function from worker library: {:?}",
-          constants::PROCESS_FUNCTION,
+          constants::INIT_PROCESS_FUNCTION,
           error
-        )),
+        ))
+      })?;
+
+    let handler = Handler {
+      job_id: None,
+      parameters: Some(parameters),
+      channel: None,
+    };
+
+    let handler_ptr = Box::into_raw(Box::new(handler));
+    let format_context_ptr = Box::into_raw(Box::new(format_context.lock().unwrap()));
+
+    let output_stream_indexes_ptr = Vec::<c_uint>::new().as_mut_ptr();
+    let return_code = init_process_func(
+      handler_ptr as *mut c_void,
+      get_parameter_value,
+      logger,
+      format_context_ptr as *mut c_void,
+      &output_stream_indexes_ptr,
+    );
+
+    if return_code == 0 {
+      let mut output_streams = vec![];
+      if !output_stream_indexes_ptr.is_null() {
+        let mut offset = 0;
+        loop {
+          let value_ptr = output_stream_indexes_ptr.offset(offset);
+          if value_ptr.is_null() {
+            break;
```

Comment:
if `is_null`, offset is not incremented. It will loop indefinitely ...

Reply:
No, it won't since if `is_null` we break the loop! :)